internal/adminapi/client.go

```diff
@@ -61,6 +61,7 @@ type KonnectClient struct {
 
 // NewKonnectClient creates an Admin API client that is to be used with a Konnect Control Plane Admin API.
 func NewKonnectClient(c *kong.Client, controlPlane string, consumersSyncDisabled bool) *KonnectClient {
+	c.SetDoer(KonnectHTTPDoer())
```

Comment:
Here, we're setting the "doer" on the provided client which is passed by pointer. This will have an effect on it in caller's context. Is that desired?

Reply:
It's intentional, but that's a good concern. I think it will be cleaner to do it one layer higher in `NewKongClientForKonnectControlPlane` which is responsible for creating the client.